File: backend/app/models/assignment.py
from sqlalchemy import Column, String, Text, Integer, ForeignKey, Enum, Float
from sqlalchemy.sql import func
from ..database import Base
from datetime import datetime
import enum

class AssignmentType(str, enum.Enum):
    HOMEWORK = "homework"
    EXAM = "exam"
    PROJECT = "project"
    QUIZ = "quiz"

class Assignment(Base):
    __tablename__ = "assignments"
    
    id = Column(String(36), primary_key=True)
    course_id = Column(String(36), ForeignKey("courses.id", ondelete="CASCADE"), nullable=False, index=True)
    module_id = Column(String(36), ForeignKey("modules.id", ondelete="SET NULL"), index=True)
    title = Column(String(255), nullable=False)
    description = Column(Text)
    assignment_type = Column(Enum(AssignmentType), default=AssignmentType.HOMEWORK)
    max_score = Column(Integer, default=100)
    due_date = Column(String(50))
    created_at = Column(String(50), default=lambda: datetime.utcnow().isoformat())

class AssignmentSubmission(Base):
    __tablename__ = "assignment_submissions"
    
    id = Column(String(36), primary_key=True)
    assignment_id = Column(String(36), ForeignKey("assignments.id", ondelete="CASCADE"), nullable=False, index=True)
    student_id = Column(String(36), ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    content = Column(Text)
    file_url = Column(String(500))
    score = Column(Float)
    feedback = Column(Text)
    submitted_at = Column(String(50), default=lambda: datetime.utcnow().isoformat())
    graded_at = Column(String(50))

File: frontend_flutter/lib/models/assignment.dart
class Assignment {
  final String id;
  final String courseId;
  final String? moduleId;
  final String title;
  final String? description;
  final String assignmentType;
  final int maxScore;
  final String? dueDate;
  final String createdAt;

  Assignment({
    required this.id,
    required this.courseId,
    this.moduleId,
    required this.title,
    this.description,
    required this.assignmentType,
    required this.maxScore,
    this.dueDate,
    required this.createdAt,
  });

  factory Assignment.fromJson(Map<String, dynamic> json) {
    return Assignment(
      id: json['id'],
      courseId: json['course_id'],
      moduleId: json['module_id'],
      title: json['title'],
      description: json['description'],
      assignmentType: json['assignment_type'],
      maxScore: json['max_score'],
      dueDate: json['due_date'],
      createdAt: json['created_at'],
    );
  }
}

class Submission {
  final String id;
  final String assignmentId;
  final String studentId;
  final String? content;
  final String? fileUrl;
  final double? score;
  final String? feedback;
  final String submittedAt;
  final String? gradedAt;

  Submission({
    required this.id,
    required this.assignmentId,
    required this.studentId,
    this.content,
    this.fileUrl,
    this.score,
    this.feedback,
    required this.submittedAt,
    this.gradedAt,
  });

  factory Submission.fromJson(Map<String, dynamic> json) {
    return Submission(
      id: json['id'],
      assignmentId: json['assignment_id'],
      studentId: json['student_id'],
      content: json['content'],
      fileUrl: json['file_url'],
      score: json['score']?.toDouble(),
      feedback: json['feedback'],
      submittedAt: json['submitted_at'],
      gradedAt: json['graded_at'],
    );
  }

  bool get isGraded => gradedAt != null;
}
